toolchain/driver/carbon_runner.h

```diff
@@ -0,0 +1,42 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_DRIVER_CARBON_RUNNER_H_
+#define CARBON_TOOLCHAIN_DRIVER_CARBON_RUNNER_H_
+
+#include <filesystem>
+
+#include "common/error.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/TargetParser/Triple.h"
+#include "toolchain/driver/codegen_options.h"
+#include "toolchain/driver/driver_env.h"
+#include "toolchain/driver/driver_subcommand.h"
+#include "toolchain/driver/tool_runner_base.h"
+#include "toolchain/lower/options.h"
+
+namespace Carbon {
+
+// Helper class for cross-subcommand functionality.
+// Allows type-safe calls of subcommands.
+// TODO: May need to be adjusted once caching is refactored.
```

Comment:
All the refactorings are now landed, so hopefully can update all of this and address the TODOs.